rta/overlayfs_privesc.py

```diff
@@ -0,0 +1,53 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0; you may not use this file except in compliance with the Elastic License
+# 2.0.
+
+from . import common
+from . import RtaMetadata
+import subprocess
+
+
+metadata = RtaMetadata(
+    uuid="a78663dc-9561-40a9-b4eb-f15e31c690cc",
+    platforms=["linux"],
+    endpoint=[{"rule_name": "Potential Privilege Escalation via OverlayFS", "rule_id": "ca9de348-a09d-4c67-af21-5645b70003d0"}],
+    siem=[{"rule_name": "Potential Privilege Escalation via OverlayFS", "rule_id": "b51dbc92-84e2-4af1-ba47-65183fcd0c57"}],
+    techniques=["T1068"],
+)
+
+@common.requires_os(metadata.platforms)
+
+def main():
+    common.log("Creating a fake unshare executable..")
+    masquerade = "/tmp/unshare"
+    source = common.get_path("bin", "linux.ditto_and_spawn")
+    common.copy_file(source, masquerade)
+
+    # Execute command
+    commands = [
+        masquerade,
+        '-rm',
+        'cap_setuid'
+    ]
+
+    common.log("Launching fake commands to set cap_setuid via unshare")
+    common.execute([*commands], timeout=2, kill=True)
+    common.log("Unshare simulation succesful")
+
+    common.log("Faking uid change via same parent")
+
+    sudo_commands = [
+        "sudo",
+        "su"
+    ]
```

Comment:
Why do we need the sudo commands?

Reply:
Usually, these RTAs are run in a root context (which basically is the point of this RTA IINM). Does the behavior still work as expected if you run on a system as the root user?

Reply:
I was not aware that these are ran as root. Would make sense though. But no, that won't work. If I want that to work, I would have to drop out of the privileged shell by running an `exit`, execute the first set of commands and then elevate again via the sub process, as the query is looking for privilege escalation activity. I could build it like that, just not sure if that wouldn't break the process of running something like `python3 -m rta all`.